utils/rrdtool1/patches/100-fix-musl122-time_t-64bit.patch

```diff
@@ -0,0 +1,171 @@
+--- a/src/rrd_tool.h
++++ b/src/rrd_tool.h
+@@ -40,6 +40,7 @@ extern "C" {
+ #include <string.h>
+ #include <time.h>
+ #include <ctype.h>
++#include <inttypes.h>
+ 
+ #if HAVE_SYS_PARAM_H
+ #  include <sys/param.h>
+--- a/src/rrd_dump.c
++++ b/src/rrd_dump.c
+@@ -39,7 +39,7 @@ rrd_dump(int argc, char **argv)
+ #else
+ # error "Need strftime"
+ #endif
+-    printf("\t<lastupdate> %ld </lastupdate> <!-- %s -->\n\n",
++    printf("\t<lastupdate> %" PRId64 " </lastupdate> <!-- %s -->\n\n",
+ 	   rrd.live_head->last_up,somestring);
+     for(i=0;i<rrd.stat_head->ds_cnt;i++){
+ 	printf("\t<ds>\n");
+--- a/src/rrd_fetch.c
++++ b/src/rrd_fetch.c
+@@ -88,7 +88,7 @@ rrd_fetch(int argc,
+     }
+     
+     if (end_tmp < start_tmp) {
+-	rrd_set_error("start (%ld) should be less than end (%ld)", start_tmp, end_tmp);
++	rrd_set_error("start (%" PRId64 ") should be less than end (%" PRId64 ")", start_tmp, end_tmp);
+ 	return(-1);
+     }
+     
+@@ -265,7 +265,7 @@ rrd_fetch_fn(
+     start_offset = (long)(*start - rra_start_time) / (long)*step;
```

Comment:
long casts for a time_t sounds wrong.

Reply:
Not my change.
I have so far only changed the printf stuff.

No compiler warnings from the current changes.

Reply:
Sure. It's an issue for upstream. Although that seems to be dead...

Reply:
But you are right, those are time_t variables.
Sad, but possibly the rrd needs more work.

(we are using year 2003 version, so we are missing all the later modifications for 64bit.)

Reply:
> Although that seems to be dead...

????
current versions: https://github.com/oetiker/rrdtool-1.x

But we are using so ancient version from 2003 that they have not even brought that to the github repo.
Our version is really small and does not require pango and cairo libraries like the new versions do.

Reply:
Ah OK.